src/compiler/binder.ts

```diff
@@ -1297,6 +1297,9 @@ namespace ts {
                     // as other properties in the object literal.  So we use SymbolFlags.PropertyExcludes
                     // so that it will conflict with any other object literal members with the same
                     // name.
+                    if (options.strictThis) {
+                        seenThisKeyword = true;
+                    }

```

Comment:
So any interface containing a method is now generic. Probably unavoidable, but we should get some data on what it means for performance.